File: backend/src/modules/userdata/userdata.controller.ts
import { Controller, Get, Post, Body, Patch, Param, Delete, Query, ParseIntPipe } from '@nestjs/common';
import { ApiTags, ApiOperation } from '@nestjs/swagger';
import { UserdataService } from './userdata.service';
import { PaginationDto } from '../../common/dto/pagination.dto';
import { ApiResponseDto } from '../../common/dto/api-response.dto';

@ApiTags('用户数据')
@Controller('api/userdata')
export class UserdataController {
  constructor(private readonly userdataService: UserdataService) {}

  // Aliases
  @Get('aliases')
  @ApiOperation({ summary: '获取别名列表' })
  async findAllAliases(@Query() dto: PaginationDto) {
    return ApiResponseDto.success(await this.userdataService.findAllAliases(dto));
  }

  @Post('aliases')
  async createAlias(@Body() data: any) {
    return ApiResponseDto.success(await this.userdataService.createAlias(data), '创建成功');
  }

  @Patch('aliases/:id')
  async updateAlias(@Param('id', ParseIntPipe) id: number, @Body() data: any) {
    return ApiResponseDto.success(await this.userdataService.updateAlias(id, data), '更新成功');
  }

  @Delete('aliases/:id')
  async removeAlias(@Param('id', ParseIntPipe) id: number) {
    await this.userdataService.removeAlias(id);
    return ApiResponseDto.success(null, '删除成功');
  }

  // Groups
  @Get('groups')
  @ApiOperation({ summary: '获取用户组列表' })
  async findAllGroups(@Query() dto: PaginationDto) {
    return ApiResponseDto.success(await this.userdataService.findAllGroups(dto));
  }

  @Post('groups')
  async createGroup(@Body() data: any) {
    return ApiResponseDto.success(await this.userdataService.createGroup(data), '创建成功');
  }

  @Patch('groups/:id')
  async updateGroup(@Param('id', ParseIntPipe) id: number, @Body() data: any) {
    return ApiResponseDto.success(await this.userdataService.updateGroup(id, data), '更新成功');
  }

  @Delete('groups/:id')
  async removeGroup(@Param('id', ParseIntPipe) id: number) {
    await this.userdataService.removeGroup(id);
    return ApiResponseDto.success(null, '删除成功');
  }

  // Speed Dial
  @Get('speed-dial')
  @ApiOperation({ summary: '获取快捷拨号列表' })
  async findAllSpeedDials(@Query() dto: PaginationDto) {
    return ApiResponseDto.success(await this.userdataService.findAllSpeedDials(dto));
  }

  @Post('speed-dial')
  async createSpeedDial(@Body() data: any) {
    return ApiResponseDto.success(await this.userdataService.createSpeedDial(data), '创建成功');
  }

  @Patch('speed-dial/:id')
  async updateSpeedDial(@Param('id', ParseIntPipe) id: number, @Body() data: any) {
    return ApiResponseDto.success(await this.userdataService.updateSpeedDial(id, data), '更新成功');
  }

  @Delete('speed-dial/:id')
  async removeSpeedDial(@Param('id', ParseIntPipe) id: number) {
    await this.userdataService.removeSpeedDial(id);
    return ApiResponseDto.success(null, '删除成功');
  }
}

File: kam-admin-module-user/src/main/java/io/kamailio/admin/modules/userdata/controller/UserdataController.java
package io.kamailio.admin.modules.userdata.controller;

import io.kamailio.admin.common.dto.ApiResponse;
import io.kamailio.admin.common.dto.PaginatedResult;
import io.kamailio.admin.common.dto.PaginationDto;
import io.kamailio.admin.modules.userdata.entity.DbAliases;
import io.kamailio.admin.modules.userdata.entity.Grp;
import io.kamailio.admin.modules.userdata.entity.SpeedDial;
import io.kamailio.admin.modules.userdata.service.UserdataService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@Tag(name = "用户数据")
@RestController
@RequestMapping("/api/userdata")
public class UserdataController {
    private final UserdataService service;

    public UserdataController(UserdataService service) { this.service = service; }

    @GetMapping("/aliases")
    @Operation(summary = "获取别名列表")
    public ResponseEntity<ApiResponse<PaginatedResult<DbAliases>>> findAllAliases(PaginationDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.findAllAliases(dto)));
    }

    @PostMapping("/aliases")
    public ResponseEntity<ApiResponse<DbAliases>> createAlias(@RequestBody DbAliases data) {
        return ResponseEntity.ok(ApiResponse.success(service.createAlias(data), "创建成功"));
    }

    @PatchMapping("/aliases/{id}")
    public ResponseEntity<ApiResponse<DbAliases>> updateAlias(@PathVariable Integer id, @RequestBody DbAliases data) {
        return ResponseEntity.ok(ApiResponse.success(service.updateAlias(id, data), "更新成功"));
    }

    @DeleteMapping("/aliases/{id}")
    public ResponseEntity<ApiResponse<Void>> removeAlias(@PathVariable Integer id) {
        service.removeAlias(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    @GetMapping("/groups")
    @Operation(summary = "获取用户组列表")
    public ResponseEntity<ApiResponse<PaginatedResult<Grp>>> findAllGroups(PaginationDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.findAllGroups(dto)));
    }

    @PostMapping("/groups")
    public ResponseEntity<ApiResponse<Grp>> createGroup(@RequestBody Grp data) {
        return ResponseEntity.ok(ApiResponse.success(service.createGroup(data), "创建成功"));
    }

    @PatchMapping("/groups/{id}")
    public ResponseEntity<ApiResponse<Grp>> updateGroup(@PathVariable Integer id, @RequestBody Grp data) {
        return ResponseEntity.ok(ApiResponse.success(service.updateGroup(id, data), "更新成功"));
    }

    @DeleteMapping("/groups/{id}")
    public ResponseEntity<ApiResponse<Void>> removeGroup(@PathVariable Integer id) {
        service.removeGroup(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    @GetMapping("/speed-dial")
    @Operation(summary = "获取快捷拨号列表")
    public ResponseEntity<ApiResponse<PaginatedResult<SpeedDial>>> findAllSpeedDials(PaginationDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.findAllSpeedDials(dto)));
    }

    @PostMapping("/speed-dial")
    public ResponseEntity<ApiResponse<SpeedDial>> createSpeedDial(@RequestBody SpeedDial data) {
        return ResponseEntity.ok(ApiResponse.success(service.createSpeedDial(data), "创建成功"));
    }

    @PatchMapping("/speed-dial/{id}")
    public ResponseEntity<ApiResponse<SpeedDial>> updateSpeedDial(@PathVariable Integer id, @RequestBody SpeedDial data) {
        return ResponseEntity.ok(ApiResponse.success(service.updateSpeedDial(id, data), "更新成功"));
    }

    @DeleteMapping("/speed-dial/{id}")
    public ResponseEntity<ApiResponse<Void>> removeSpeedDial(@PathVariable Integer id) {
        service.removeSpeedDial(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }
}
